modules/fedimint-lnv2-client/src/send_sm.rs

```diff
@@ -43,10 +43,23 @@ pub struct SendSMCommon {
     pub funding_txid: TransactionId,
     pub gateway_api: SafeUrl,
     pub contract: OutgoingContract,
-    pub invoice: Bolt11Invoice,
+    pub invoice: Invoice,
     pub refund_keypair: KeyPair,
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Decodable, Encodable)]
+pub enum Invoice {
```

Comment:
Maybe this should go in the common crate?

Reply:
I would not share this for cleaner separation.

Reply:
Why is this cleaner? I see repeated code here and in the gateway client